http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -1291,4 +1323,40 @@ public RequestDistribution(HttpClient httpClient, Processor processor)
             });
         }
     }
+
+    // By wrapping HttpClient, we are able to substitute the underlying AuthenticationStore
+    // with a more efficient one.
+    private static class WrappedHttpClient
+            extends HttpClient
+    {
+        private final AuthenticationStore authenticationStore;
+
+        public WrappedHttpClient(ClientSecurityConfig securityConfig, HttpClientTransport transport,
+                SslContextFactory sslContextFactory)
+        {
+            super(transport, sslContextFactory);
+            if (securityConfig.getAuthenticationEnabled()) {
+                requireNonNull(securityConfig.getKrb5ConfPath(), "krb5ConfPath is null");
+                requireNonNull(securityConfig.getServiceName(), "serviceName is null");
+                authenticationStore = new ExpiringAuthenticationStore(new SpnegoAuthentication(
+                        securityConfig.getKrb5ConfPath(), securityConfig.getServiceName()));
+            }
+            else {
+                authenticationStore = super.getAuthenticationStore();
+            }
+        }
+
+        @Override
+        public AuthenticationStore getAuthenticationStore() {
```

Comment:
Formatting: brace needs to go on the next line - see https://github.com/airlift/codestyle.